server/slash_commands.go

```diff
@@ -0,0 +1,226 @@
+			}
+			s.addLabel(pr.RepoOwner, pr.RepoName, pr.Number, label)
+		},
+		updateHandler: func(envMap cloudModel.EnvVarMap) {
+			spinwick := model.NewSpinwick(pr.RepoName, pr.Number, s.Config.DNSNameTestServer)
+			s.envMapsLock.Lock()
+			s.envMaps[spinwick.RepeatableID] = envMap
+			s.envMapsLock.Unlock()
+
+			s.handleSynchronizeSpinwick(pr, spinwick.RepeatableID, true)
+		},
+		deleteHandler: func() {
+			for _, label := range pr.Labels {
+				if s.isSpinWickLabel(label) {
+					s.removeLabel(pr.RepoOwner, pr.RepoName, pr.Number, label)
+				}
+			}
+		},
+	}
+
+	switch args[0] {
+	case slashCommandSpinWick:
+		output, err := s.handleSpinWickSlashCommand(args[1:], spinWickHandlers)
+		if err != nil {
+			s.Logger.WithError(err).Error("failed to handle spinwick command")
+		}
+		if output != "" {
+			s.sendGitHubComment(ev.GetRepo().GetOwner().GetLogin(),
+				ev.GetRepo().GetName(),
+				ev.GetIssue().GetNumber(), fmt.Sprintf("```\n%s\n```", output))
+		}
+	case slashCommandShrugWick:
+		s.handleShrugWick(ev)
+	default:
+		s.Logger.WithField("cmd", cmd).Error("invalid slash command")
+	}
+}
+
+func (s *Server) parseSpinwickSlashCommandArgs(args []string, isUpdate bool) (spinWickSlashCommandArgs, string, error) {
+	var parsedArgs spinWickSlashCommandArgs
+
+	var outBuf bytes.Buffer
+	flagset := flag.NewFlagSet("spinwick", flag.ContinueOnError)
+	flagset.SetOutput(&outBuf)
+
+	var env string
+	var clearEnv string
+	var size string
+	flagset.StringVar(&env, "env", "", "An optional comma-separated list of environment variables. Example: VAR1=VAl1,VAR2=VAL2")
+	if isUpdate {
+		flagset.StringVar(&clearEnv, "clear-env", "", "An optional comma-separated list of environment variables to clear. Example: VAR1,VAR2")
+	} else {
+		flagset.StringVar(&size, "size", "miniHA", "Size of the Mattermost installation e.g. 'miniSingleton' or 'miniHA'")
+	}
+
+	err := flagset.Parse(args)
+	if errors.Is(err, flag.ErrHelp) {
+		return parsedArgs, outBuf.String(), err
+	} else if err != nil {
+		return parsedArgs, outBuf.String(), fmt.Errorf("failed to parse args: %w", err)
+	}
+
+	s.Logger.WithField("env", env).Info("parsed env vars")
+
+	envMap := make(cloudModel.EnvVarMap)
+	if env != "" {
+		envMap, err = parseEnvArg(env)
+		if err != nil {
+			return parsedArgs, err.Error(), fmt.Errorf("failed to parse env vars: %w", err)
+		}
+	}
+
+	if clearEnv != "" {
+		keys := splitCommaSeparated(clearEnv)
+		for _, k := range keys {
+			envMap[k] = cloudModel.EnvVar{}
+		}
+	}
+
+	parsedArgs.envMap = envMap
+	parsedArgs.size = size
+
+	return parsedArgs, "", nil
+}
+
+var spinwickSlashCommandUsageString = `Usage: /spinwick <command> [args]
+
+Available commands:
+  create  Create a new Mattermost spinwick installation
+  update  Update the existing Mattermost spinwick installation
+  delete  Delete the existing Mattermost spinwick installation
+`
+
+func (s *Server) handleSpinWickSlashCommand(args []string, handlers spinWickSlashCommandsHandlers) (string, error) {
```

Comment:
Optional suggestion:

You could refactor this command structure to use cobra.Command. That would simplify the code here quite a bit by covering the argument parsing, command selection, automate help text etc. It would also open up the ability to use StringVarSlice for parsing the env variables

This type of refactor would probably be right up Aider's alley :)

Reply:
I generally prefer a little verbosity if it means standard library can be used in place of a new dependency but let me think about this one.